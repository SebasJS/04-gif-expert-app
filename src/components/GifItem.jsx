import PropTypes from 'prop-types'

const GifItems = ( { title, id, url }) => {
    
    return (
        <div className="card">
            <img src={ url } alt={ title } />    
            <p>{ title }</p>
        </div>)
}

GifItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

export default GifItems;