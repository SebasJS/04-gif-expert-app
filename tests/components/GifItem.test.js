import { render, screen } from '@testing-library/react'
import GifItem from '../../src/components/GifItem'



describe('Prueba de componente GifItem.jsx', () => {

    const title = 'Saitama'
    const url = 'https://one-punch.com/saitama'
    test('Debe hacer match un snapshot de GifItem', () => {
        
        const { container } = render( <GifItem title={ title } url={ url }/>);
        expect(container).toMatchSnapshot();
    });
    test('debe de mostrar el URL y el ALT inidicado', () => {
        render( <GifItem title={ title } url={ url }/>);
        // expect( screen.getByRole('img').src).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );

    });
    test('debe de mostrar el titulo', () => {
        
        render( <GifItem title={ title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy();

    });

});