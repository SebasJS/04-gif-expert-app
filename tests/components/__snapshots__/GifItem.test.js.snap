// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifItem.jsx Debe hacer match un snapshot de GifItem 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
